src/components/encoding-pane/function-picker.tsx

```diff
@@ -19,17 +25,29 @@ export class FunctionPickerBase extends React.PureComponent<FunctionPickerProps,
     this.onFunctionChange = this.onFunctionChange.bind(this);
   }
   public render() {
-    const {fieldDef} = this.props;
-
-    const fn = fieldDef.aggregate || fieldDef.timeUnit || (fieldDef.bin && 'bin' ) || undefined;
+    const {fieldDef, filter, index} = this.props;
+    let fn: any, name: string;
+    if (filter) {
+      fn = filter.timeUnit;
+      name = index.toString();
+    } else {
+      fn = fieldDef.aggregate || fieldDef.timeUnit || (fieldDef.bin && 'bin' ) || undefined;
+      name = JSON.stringify(fieldDef);
```

Comment:
I assume that we need `name` otherwise it is buggy.  

If so, if filter uses index, why don't we use shelfId here? 
I think maybe instead of making both `index` and `shelfId` a parameter, why don't we make name a parameter of FunctionPicker?